### ISSUE TITLE
Autoinstall not Working on Ubuntu 22.04 and 23.10

### ISSUE THREAD

**Issue by BRVN01** [canonical/cloud-init#4304]
# Bug report
I'm facing an issue while trying to install Ubuntu 22.04 using Autoinstall. It seems that Ubuntu is either not reading the user-data file or encountering an error without providing any specific details, even though the user-data works on another ISO of the same Ubuntu version.

During the process of loading the user-data configurations for Autoinstall, the installer simply skips to the language configuration step, and the installation proceeds manually.

I have two ISOs of Ubuntu Server 22.04, where the installation works on one but not on the other (using the same user-data). This indicates that the issue is not with the user-data file itself (I think so).

The working version is identified by the following command:
```bash
$ cat /srv/pxe_nfs/ubuntu2204_20230306/.disk/info 
Ubuntu-Server 22.04.2 LTS "Jammy Jellyfish" - Release amd64 (20230306)
```

While the non-working version is identified as:
```bash
$ cat /srv/pxe_nfs/ubuntu2204_20230728/.disk/info 
Ubuntu-Server 22.04.2 LTS "Jammy Jellyfish" - Release amd64 (20230728)
```

Interestingly, I was able to make it work by copying the 'ubuntu-server-minimal.squashfs' and 'ubuntu-server-minimal.squashfs.gpg' files from the working version to the non-working one. I believe this might provide a clue to pinpoint the problem. I had the same issue when testing Ubuntu 23.10, where the automatic installation also didn't work (I haven't tested if the described solution works for it yet).

Here's the command I used to solve it:
```bash
$ cp /srv/pxe_nfs/ubuntu2204_20230306/casper/{ubuntu-server-minimal.squashfs,ubuntu-server-minimal.squashfs.gpg} /srv/pxe_nfs/ubuntu2204_20230728/casper/
```

I got the ISO from the following link:
https://cdimage.ubuntu.com/ubuntu-server/jammy/daily-live/current/jammy-live-server-amd64.iso

Grub line:
```bash
menuentry 'Install Ubuntu 22.04 20230728 nocloud-net' {
        set gfxpayload=keep
        linux kernels/ubuntu2204_20230728/vmlinuz ip=dhcp netboot=nfs boot=casper nfsroot=192.168.100.4:/srv/pxe_nfs/ubuntu2204_20230728/ autoinstall ds="nocloud;s=/cdrom/nocloud/" ---
        initrd kernels/ubuntu2204_20230728/initrd
}
```

## Steps to reproduce the problem
```bash
# Download the Ubuntu ISO:
wget https://cdimage.ubuntu.com/ubuntu-server/jammy/daily-live/current/jammy-live-server-amd64.iso

# Mount the ubuntu ISO:
sudo mount -r jammy-live-server-amd64.iso /mnt

# Create folder to put `user-data` and `meta-data`?
mkdir -p ~/www
cd ~/www

cat > user-data << 'EOF'
#cloud-config
autoinstall:
  version: 1
  identity:
    hostname: ubuntu-server
    password: "$6$exDY1mhS4KUYCE/2$zmn9ToZwTKLhCw.b4/b.ZRTIZM30JZ4QrOQ2aOXJ8yk96xpcCof0kxKwuX1kqLG/ygbJ1f8wxED22bTL4F46P0"
    username: ubuntu
EOF

touch meta-data

# With another terminal run the commands below:
cd ~/www
python3 -m http.server 3003

# Create a target disk:
truncate -s 10G image.img

# Run the install:
kvm -no-reboot -m 2048 \
    -drive file=image.img,format=raw,cache=none,if=virtio \
    -cdrom jammy-live-server-amd64.iso \
    -kernel /mnt/casper/vmlinuz \
    -initrd /mnt/casper/initrd \
    -append 'root=/dev/ram0 ramdisk_size=1500000 ip=dhcp autoinstall ds=nocloud-net;s=http://_gateway:3003/'
```
If you use old Ubuntu 22.04 these steps will work!

## Environment details
- Cloud-init version: One ISO has `22.4.2-0ubuntu0~22.04.1` and another `23.2.1-0ubuntu0~22.04.1`.
- Operating System Distribution: Ubuntu 22.04.2 LTS
- Cloud provider, platform or installer type: KVM and I believe bare metal has the same problem.

## cloud-init logs
This problem does not generate any logs!

**Comment by blackboxsw** [comment]
Thank you for the bug and making cloud-init better. Given that this is autoinstall user-data and cloud-init 23.2.1, I'm presuming this is a duplicate of https://github.com/canonical/cloud-init/issues/4158 which is not a cloud-init bug, but an Ubuntu live installer bug. I expect on your system you would see a Traceback in /var/log/installler/subiquity-server-debug.log. Note this is is a known problem due to a recent SRU of cloud-init 23.2.1 to Jammy that was just fixed last week in the ubuntu live installer (subiquity) https://bugs.launchpad.net/subiquity/+bug/2022102/comments/5 as well. If this is not a symptom of jammy daily that isn't resolved in a couple of days. Feel free to open a new subiquity bug or re-open this cloud-init bug.

**Comment by dbungert** [comment]
A fix for this issue has been included in the daily builds of what will be 22.04.3.  Please use build 20230801.1 or newer.
